src/libraries/tests.proj

```diff
@@ -41,8 +41,19 @@
   <Import Project="$(RepositoryEngineeringDir)coverage.targets" Condition="'$(EnableCoverageSupport)' == 'true'" />
   <Import Sdk="Microsoft.NET.Sdk" Project="Sdk.targets" />
   
+  <ItemGroup>
+    <SetupProjects Include="$(MSBuildThisFileDirectory)restore\runtime\runtime.depproj" />
```

Comment:
This is CI only, why not directly invoking the runtime.depproj before tests.proj?

Reply:
one issue is that build parallelization might cause a race, pretest.proj has to complete before tests start to build, since pretest generates the dep file to ensure testhost is setup correctly.

Reply:
Not following. You can do two steps before the test build so the sequence would be:
1. runtime.proj
2. pretest.proj and
3. tests.proj

Noteworthy, if you would wait until Monday, we could just invoke the lib-pretest lib-tests subsets which would handle all that for you. I plan to merge this on Monday if ready: https://github.com/dotnet/runtime/pull/33553.

Reply:
I ideally want to merge this tomorrow, since this PR touches a bunch of infra pieces. I can commit to reworking this once your PR is merged.

Reply:
No problem at all, I can make the change afterwards.